src/main/java/pl/javastart/task/Firma.java

```diff
@@ -14,26 +20,54 @@ public void wyswietlPodsumowanie() {
         double sumaPrzychodow = zsumujPrzychody();
 
         System.out.printf("======= Firma: %s ===========\n", nazwa);
-        System.out.printf("Forma opodatkowania: %s\n", "TODO");
+        System.out.printf("Forma opodatkowania: %s\n", formaOpodatkowania.opodatkowanieInfo());
         System.out.printf("Suma przychodów: %.2f zł\n", sumaPrzychodow);
         System.out.printf("Suma wydatków: %.2f zł\n", zsumujWydatki());
-        System.out.printf("Podatek do zapłacenia: %.2f zł", formaOpodatkowania.wyliczPodatek(sumaPrzychodow));
+        if (formaOpodatkowania instanceof PodatekLiniowy || formaOpodatkowania instanceof SkalaPodatkowa) {
+            System.out.printf("Podatek do zapłacenia: %.2f zł", formaOpodatkowania.wyliczPodatek(obliczDochod()));
+        } else {
+            System.out.printf("Podatek do zapłacenia: %.2f zł", formaOpodatkowania.wyliczPodatek(sumaPrzychodow));
+        }
         System.out.print("\n\n");
     }
 
     private double zsumujWydatki() {
-        return 0;
+        double sumaWydatkow = 0;
+        for (double wydatek: wydatki) {
+            sumaWydatkow += wydatek;
+        }
+        return sumaWydatkow;
     }
 
     private double zsumujPrzychody() {
-        return 0;
+        double sumaPrzychodow = 0;
+        for (double przychod: przychody) {
```

Comment:
Checkstyle zauważył w tej linii błąd. To, co znalazł możesz podejrzeć w pull requeście z zakładce "Files changed". Polecam zainstalować sobie dodatek do intellij, żeby na bieżąco informował. Instrukcję znajdziesz tutaj https://bootcamp.javastart.pl/strona/jakosc-kodu